boards/arm/nrf5340dk_nrf5340/Kconfig.defconfig

```diff
@@ -14,6 +14,17 @@ config BOARD
 config BUILD_WITH_TFM
 	default y if BOARD_NRF5340DK_NRF5340_CPUAPPNS
 
+if BUILD_WITH_TFM
+
+# By default, if we build with TF-M, instruct build system to
+# flash the combined TF-M (Secure) & Zephyr (Non Secure) image
+# (when building in-tree tests).
+config TFM_FLASH_MERGED_BINARY
+	bool
+	default y if TEST_ARM_CORTEX_M
```

Comment:
Would it be cleaner if we have `depends on BUILD_WITH_TFM` instead of the surrounding `if` ? Since its only one option inside the `if`

Reply:
Could we have the comment in a help text? IMO there is no reason not to, and a user might look up the definition if debugging some issue in kconfig.

Reply:
> Would it be cleaner if we have depends on BUILD_WITH_TFM instead of the surrounding if ? Since its only one option inside the if

I recall that @ulfalizer did not like that; it's cleaner with external if, endif, and also scallable, in case we're going to add more options. No strong opinion, though, it's sort of cosmetic

Reply:
> Could we have the comment in a help text? IMO there is no reason not to, and a user might look up the definition if debugging some issue in kconfig.

Not sure about that. Kconfig.defconfig is supposed to override default options of already defined Kconfig symbols, so adding the comment as help text is confusing (the option already has a help text where it is defined in the first place).

The point of the comment here is to explain _why_ we need to override.

Reply:
> Kconfig.defconfig is supposed to override default options of already defined Kconfig symbols, so adding the comment as help text is confusing

Exactly. So a Kconfic.defconfig is solely for overriding, and should not provide extra help text or prompts (unless original symbol is promptless and a prompt is now needed, but that is special case).